syntax = "proto3";

service Booking {
    rpc GetListBookings(Empty) returns (stream BookingData) {}
    rpc GetBookingByID(BookingID) returns (BookingData) {}
    rpc AddBooking(BookingData) returns (BookingData) {}
    rpc UpdateBooking(BookingData) returns (BookingData) {}
    rpc DeleteBooking(BookingID) returns (BookingData) {}
}

message BookingID {
    string id = 1;
}

message Movie {
    string id = 1;
}

message Schedule {
    string date = 1;
    repeated Movie movies = 2;
}

message BookingData {
    string userId = 1;
    repeated Schedule schedules = 2;
}

message Empty {}
